src/screens/studentSettings/SlideCardSwitch.tsx

```diff
@@ -16,13 +16,13 @@ export class SlideCardSwitch extends React.PureComponent<Props, State> {
   constructor(props: any) {
     super(props);
     this.state = {
-      switch: true,
+      switch: props.student.slideCardSwitch,
     };
   }
 
   toggleSwitch = () => {
+    this.props.student.update({ slideCardSwitch: !this.state.switch });
```

Comment:
I think we should use the callback argument of state here. Nice description of this concept is in the "setState is async!!" section of this article:
https://www.freecodecamp.org/news/get-pro-with-react-setstate-in-10-minutes-d38251d1c781/

Reply:
Thank, I updated the code.